packages/react-events/src/__tests__/Swipe-test.internal.js

```diff
@@ -0,0 +1,326 @@
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 50, 50));
+      expect(onSwipe).toHaveBeenCalledTimes(1);
+    });
+
+    it('ignores emulated "mousemove" event', () => {
+      ref.current.dispatchEvent(createTouchEvent('touchstart'));
+      ref.current.dispatchEvent(createTouchEvent('touchend'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 50, 50));
+      ref.current.dispatchEvent(createMouseEvent('pointerup'));
+      expect(onSwipe).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Swipe Directions', () => {
+    let ref, directions;
+
+    beforeEach(() => {
+      directions = [];
+      function onSwipe(e) {
+        directions.push(e.direction);
+      }
+
+      ref = React.createRef();
+      const element = (
+        <Swipe onSwipe={onSwipe}>
+          <div ref={ref} />
+        </Swipe>
+      );
+      ReactDOM.render(element, container);
+    });
+
+    it('should be right', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['right']);
+    });
+    it('should be left', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', -50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['left']);
+    });
+    it('should be up', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 50));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['up']);
+    });
+    it('should be down', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, -50));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['down']);
+    });
+
+    it('directions in events in the correct order vertical', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', -50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 50, 0));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 60, 0));
+      ref.current.dispatchEvent(createMouseEvent('mouseup'));
+      expect(directions).toEqual(['left', 'right', 'right']);
+    });
+    it('directions in events in the correct order horizontal', () => {
+      ref.current.dispatchEvent(createMouseEvent('mousedown'));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, -50));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 50));
+      ref.current.dispatchEvent(createMouseEvent('mousemove', 0, 60));
```

Comment:
There are no tests that assert for events that have both a horizontal and vertical direction